R/processCapability.R

```diff
@@ -16,35 +16,57 @@
 #
 
 processCapability <- function(jaspResults, dataset, options){
-  
-  diameter <- unlist(options$diameter)
+
+variables <- unlist(options$diameter)
   subgroupsName <- options$subgroups
   makeSubgroups <- subgroupsName != ""
-  
+
   if (is.null(dataset)) {
     if (makeSubgroups) {
-      dataset         <- .readDataSetToEnd(columns.as.numeric = diameter, columns.as.factor = subgroupsName)
-      dataset.factors <- .readDataSetToEnd(columns = diameter, columns.as.factor = subgroupsName)
+      dataset         <- na.omit(.readDataSetToEnd(columns.as.numeric = variables, columns.as.factor = subgroupsName))
+      dataset.factors <- na.omit(.readDataSetToEnd(columns = variables, columns.as.factor = subgroupsName))
```

Comment:
there's an argument for this already
```suggestion
      dataset         <- .readDataSetToEnd(columns.as.numeric = variables, columns.as.factor = subgroupsName, exclude.na.listwise = c(variables, subgroupsName))
      dataset.factors <- .readDataSetToEnd(columns = variables, columns.as.factor = subgroupsName, exclude.na.listwise = c(variables, subgroupsName))
```